crates/bevy_reflect/src/list.rs

```diff
@@ -223,38 +223,38 @@ impl DynamicList {
     }
 
     /// Appends a typed value to the list.
-    pub fn push<T: Reflect>(&mut self, value: T) {
+    pub fn push<T: PartialReflect>(&mut self, value: T) {
         self.values.push(Box::new(value));
     }
 
     /// Appends a [`Reflect`] trait object to the list.
```

Comment:
Some docs like this still use `Reflect` instead of `PartialReflect`. It's a minor thing that we don't have to dedicate to fixing in this PR, but we should probably do a pass over the documentation in a followup PR (or at least make an issue for it).

Reply:
this looks like a mistake on my part. i reckon this should bind to `Reflect` as per the PR description recommendations.